src/index.ts

```diff
@@ -6,29 +6,88 @@ import { IThemeManager } from '@jupyterlab/apputils';
-  id: 'jupyter-theme-editor:plugin',
+  id: PLUGIN_ID,
   autoStart: true,
   requires: [IThemeManager],
-  activate: async (app: JupyterFrontEnd, themeManager: IThemeManager) => {
-    const onThemeChanged = (
-      themeManager: IThemeManager,
-      changes: IChangedArgs<string, string | null, string>
-    ) => {
+  optional: [ISettingRegistry, ITranslator],
+  activate: (
+    app: JupyterFrontEnd,
+    themeManager: IThemeManager,
+    settingRegistry: ISettingRegistry,
+    translator: ITranslator
+  ) => {
+    const onThemeChanged = (themeManager: IThemeManager) => {
       themeManager.themeChanged.disconnect(onThemeChanged);
       const model = new ThemeEditorModel();
-      const view = new ThemeEditorView(model);
+      const view = new ThemeEditorView(model, translator);
       view.addClass('jp-theme-editor-view-panel');
       view.id = 'theme-editor';
       view.title.icon = themeEditorIcon;
+      model.schema = {};
       app.shell.add(view, 'left');
+      console.log('JupyterLab extension jupyter-theme-editor is activated!');
+
+      Promise.all([app.restored, settingRegistry.load(PLUGIN_ID)]).then(
+        ([, setting]) => {
+          model.schema = setting.schema.properties?.customStyles ?? {};
+          model.stateChanged.emit();
+          let useSettings: any = null;
+          let customStyles: any = {};
+          const onModelChanged = () => {
+            for (const key of Object.keys(model.formData)) {
+              if (typeof model.formData[key] === 'string') {
+                model.formData[key] = model.formData[key].trim();
+              }
+            }
+            // Expanding will result in a copy
+            customStyles = { ...model.formData };
+            setting.set('customStyles', customStyles);
+          };
+
+          loadSetting();
+          setting.changed.connect(loadSetting);
+
+          function loadSetting() {
+            // Read the settings and convert to the correct type
+            const candidate = setting.get('useSettings').composite as boolean;
+            customStyles = setting.get('customStyles').composite as any;
+            if (candidate !== useSettings) {
+              useSettings = candidate;
+
+              // ... apply needed change
+              if (useSettings === false) {
+                model.stateChanged.connect(onModelChanged);
+              } else {
+                model.stateChanged.disconnect(onModelChanged);
+                model.formData = { ...customStyles };
+                model.stateChanged.emit();
+                model.stateChanged.connect(onModelChanged);
+              }
+            }
+          }
+        }
+      );
     };
     themeManager.themeChanged.connect(onThemeChanged);
+    requestAPI<any>('get_example')
+      .then(data => {
+        console.log(data);
+      })
+      .catch(reason => {
+        console.error(
+          `The jupyter_theme_editor server extension appears to be missing.\n${reason}`
+        );
+      });
```

Comment:
This seems like an error induced by the rebase?

Reply:
@fcollonval Thanks. It was indeed an error induced by the rebase. Sorry for that. Now all checks have passed.